pkg/resource/plugin/analyzer_plugin.go

```diff
@@ -361,6 +415,117 @@ func marshalProvider(provider *AnalyzerProviderResource) (*pulumirpc.AnalyzerPro
 	}, nil
 }
 
+func marshalEnforcementLevel(el apitype.EnforcementLevel) pulumirpc.EnforcementLevel {
+	switch el {
+	case apitype.Advisory:
+		return pulumirpc.EnforcementLevel_ADVISORY
+	case apitype.Mandatory:
+		return pulumirpc.EnforcementLevel_MANDATORY
+	case apitype.Disabled:
+		return pulumirpc.EnforcementLevel_DISABLED
+	}
+	contract.Failf("Unrecognized enforcement level %s", el)
+	return 0
+}
+
+func marshalMap(m map[string]interface{}) *structpb.Struct {
```

Comment:
A shame we have to write all this - isn't there something already that does this?

Reply:
Yes, it's unfortunate there are no built-in helpers for this. https://github.com/golang/protobuf/issues/370